### ISSUE TITLE
Plugin interface for storing scan state

### ISSUE THREAD

**Issue by stanislaw55** [sardana-org/sardana#1382]
Hi,
it would be nice to have plugin interface for storing Sardana's scan data in some other place then process memory.
Example: there's some long running scan. At some point, Sardana hangs or crashes - no matter what caused it, either Sardana itself or some poorly written plugin. Scan data is preserved in form of files or other provided by Recorders. However, scan cannot be easily restored - external state storage would allow easy resuming scan when Sardana crashed or hung up. Plugin interface would allow to do so. It would be beneficial also in case of overall Sardana stability.

**Comment by cpt-majkel** [comment]
We think that plugin interface is not necessary here. It could be achieved by using Macroserver environment. We just need to clarify if this is just in case of hangs or crashes or also for exceptions during scans. Implemetation could be done by adding a new macro eg. `continue_last_scan` that would take info from this environment variable and continue the scan.

**Comment by cpascual** [comment]
@cpt-majkel's suggestion LGTM, but whatever the implementation (external persistency system via plugins or just using the MS env), I think that the key discussion is how fine-grained the "restore info" would be (i.e., how often can we store the state without introducing too much performance penalty)

**Comment by reszelaz** [comment]
I agree with @cpascual. @cpt-majkel could you please try to update this issue with this information?
